File: src/server/api/go/internal/modules/model/experience_confirmation.go
package model

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/datatypes"
)

type ExperienceConfirmation struct {
	ID             uuid.UUID         `gorm:"type:uuid;default:gen_random_uuid();primaryKey" json:"id"`
	SpaceID        uuid.UUID         `gorm:"type:uuid;not null;index:idx_experience_confirmations_space" json:"space_id"`
	ExperienceData datatypes.JSONMap `gorm:"type:jsonb;not null" swaggertype:"object" json:"experience_data"`

	CreatedAt time.Time `gorm:"autoCreateTime;not null;default:CURRENT_TIMESTAMP" json:"created_at"`
	UpdatedAt time.Time `gorm:"autoUpdateTime;not null;default:CURRENT_TIMESTAMP" json:"updated_at"`

	// ExperienceConfirmation <-> Space
	Space *Space `gorm:"foreignKey:SpaceID;references:ID;constraint:OnDelete:CASCADE,OnUpdate:CASCADE;" json:"-"`
}

func (ExperienceConfirmation) TableName() string { return "experience_confirmations" }

File: src/server/core/acontext_core/schema/orm/experience_confirmation.py
from dataclasses import dataclass, field
from sqlalchemy import ForeignKey, Index, Column
from sqlalchemy.orm import relationship
from sqlalchemy.dialects.postgresql import UUID, JSONB
from typing import TYPE_CHECKING

from .base import ORM_BASE, CommonMixin
from ..utils import asUUID

if TYPE_CHECKING:
    from .space import Space


@ORM_BASE.mapped
@dataclass
class ExperienceConfirmation(CommonMixin):
    __tablename__ = "experience_confirmations"

    __table_args__ = (Index("idx_experience_confirmations_space", "space_id"),)

    space_id: asUUID = field(
        metadata={
            "db": Column(
                UUID(as_uuid=True),
                ForeignKey("spaces.id", ondelete="CASCADE", onupdate="CASCADE"),
                nullable=False,
            )
        }
    )

    experience_data: dict = field(metadata={"db": Column(JSONB, nullable=False)})

    # Relationships
    space: "Space" = field(
        init=False,
        metadata={
            "db": relationship(
                "Space",
                back_populates="experience_confirmations",
            )
        },
    )
